gravitee-am-service/src/main/java/io/gravitee/am/service/validators/flow/policy/SendEmailPolicyValidator.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.service.validators.flow.policy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gravitee.am.model.flow.Step;
+import io.gravitee.am.service.validators.email.EmailDomainValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Component
+public class SendEmailPolicyValidator implements PolicyValidator {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(SendEmailPolicyValidator.class);
+    public static final String POLICY_AM_SEND_EMAIL = "policy-am-send-email";
+    public static final String FROM_KEY = "from";
+    private final EmailDomainValidator emailDomainValidator;
+    private final ObjectMapper objectMapper;
+
+    public SendEmailPolicyValidator(EmailDomainValidator emailDomainValidator, ObjectMapper objectMapper) {
+        this.emailDomainValidator = emailDomainValidator;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public Boolean validate(Step element) {
+        if (POLICY_AM_SEND_EMAIL.equalsIgnoreCase(element.getPolicy())) {
+            try {
+                var tree = objectMapper.readTree(element.getConfiguration());
+                final String from = ofNullable(tree.get(FROM_KEY)).map(JsonNode::textValue).orElse(null);
+                return emailDomainValidator.validate(from);
+            } catch (Exception e) {
+                LOGGER.debug("An unexpected error has occurred", e);
```

Comment:
If something unexpected happens, we want to know about it. I would probably log this with `error` level

Reply:
Agree, should be at least a WARN

Reply:
Let's go with this then